students/ahwall/Lesson06Homework/Lesson06HW_Git/MailRoomPt3.py

```diff
@@ -0,0 +1,167 @@
+        thank_you()
+        return list_of_donors
+    else:
+        i = 0
+        while i < 1:
+            for item in donors:
+                DonorInList = False
+                if item[0].count(ty_letter) == 1:
+                    DonorInList = True  # works
+                    break
+            if DonorInList is True:
+                j = 0
+                for item in donors:
+                    if ty_letter == item[0]:  # Works
+                        try:
+                            DonationVal = input("\nDonor found,\n\nHow much was \
+their donate?  ")
+                            donorprof = donors[j]
+                            vallist = donorprof[1]
+                            DonationVal = float(DonationVal)
+                        except ValueError:
+                            return ("\nYou did not enter an numberic value, Please \
+try again\n")
+                            thank_you()
+                        else:
+                            vallist.append(DonationVal)
+                            print()
+                            return (ty_letter, "\b's donation has been added to \
+the Donor Report...Creating 'Thank you email'")
+                            i = 1
+                            One_letter(ty_letter, DonationVal)
+                            mainloop()
+                            break
+                    else:
+                        j += 1
+            else:  # WORKS
+                try:
+                    DonationVal = input("\nAdding Donor to list,\n\nHow much was \
+their donate?  ")
+                #  place exception to make sure float is given
+                    DonationVal = float(DonationVal)
+                    vallist = []
+                    vallist.append(DonationVal)
+                except ValueError:
+                    return ("\nYou did not enter an numberic value, Please try \
+again\n")
+                    # thank_you()
+                else:
+                    Don_Val = ()
+                    Don_Val = (ty_letter[:], vallist[:])
+                    donors.append(Don_Val)
+                    print()
+                    return (ty_letter, "and their donation have been added to \
+the Donor Report...Creating 'Thank you email'")
+                    i = 1
+                    One_letter(ty_letter, DonationVal)
+                    mainloop()
+                break  # Works
+
+
+def One_letter(donor, donation):
+    print()
+    return "Dear {};\n\n\tThank you for your generous donation of ${:.2f}.\n\n \
+\tSincerely,\n\n \tThe Python Project\n".format(donor, donation)  # Works
+
+
+def All_letters(donordict):
+    donorvallist = list(donordict.values())
+    for i in range(len(donorvallist[0])):
+        Donor, Donation = donordict['Donor'][i], donordict['Donation'][i]
+        Donation = int(Donation)
+        # could not figure out a way to unpack the dictionary (like Mr. Barker
+        # suggested in the question) with multiple values per key, since you
```

Comment:
Ohh yeah, this'll require you to change your dictionary structure to something like:
```
donor_dict{
"Donor 1": [21.45, 23.64, 2.67]
"Donor 2": [2.45, 23.64, 2.67]
}
```